src/bpfbytecode.cpp

```diff
@@ -22,6 +23,14 @@ BpfBytecode::BpfBytecode(const void *elf, size_t elf_size)
     auto map = BpfMap(m);
     maps_.emplace(name, map);
   }
+
+  struct bpf_program *p;
+  bpf_object__for_each_program(p, bpf_object_.get())
+  {
+    std::string name = bpf_program__name(p);
+    auto prog = BpfProgram(p);
+    programs_.emplace(name, std::move(prog));
```

Comment:
nit: I think this should construct BpfProgram in-place `programs_.emplace(name, p);`.